aws/resource_aws_cognito_user_pool_client.go

```diff
@@ -0,0 +1,215 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoUserPoolClient() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoUserPoolClientCreate,
+		Read:   resourceAwsCognitoUserPoolClientRead,
+		Update: resourceAwsCognitoUserPoolClientUpdate,
+		Delete: resourceAwsCognitoUserPoolClientDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"generate_secret": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"explicit_auth_flows": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateCognitoUserPoolClientAuthFlows,
+				},
+			},
+
+			"read_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"write_attributes": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
```

Comment:
Likewise - ^

Reply:
`write_attributes` is `ForceNew: false` but I guess the redundant code can be removed

Reply:
Correct.